### PR TITLE
Setup for pyproject

### PR DESCRIPTION
**Pull request by tanmayy24**
This PR updates the pyproject.toml file and other associated setup files to modernize the build configuration for the mirdata project.

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -12,7 +12,6 @@ build:
 # Build documentation in the docs/ directory with Sphinx
 sphinx:
   configuration: docs/conf.py
-  fail_on_warning: true
```
Comment (guillemcortes) [inline .readthedocs.yaml:15]: We don't want this anymore?

Comment (tanmayy24) [reply .readthedocs.yaml:15]: I made it how we put it in soundata

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11"
+]
+dependencies = [
+    "librosa>=0.10.0",
+    "numpy>=1.21.6,<2.0",
+    "pandas>=1.3.5",
+    "tqdm>=4.65.0",
+    "jams>=0.3.4",
+    "py7zr>=0.16.0"
```
Comment (genisplaja) [inline pyproject.toml:30]: I can see that several dependencies are removed form this list. I guess the installation still works as expected @tanmayy24?

Comment (tanmayy24) [reply pyproject.toml:30]: yes, I checked using 
pip install .
pip install ."[tests]"
pip install ."[docs]"

Comment (tanmayy24) [reply pyproject.toml:30]: Yes tested using
pip install .
pip install ."[tests]"
pip install ."[docs]"

**Review by genisplaja — APPROVED** [review]
lgtm then :)

**Pull request closed without merging**